importlib_metadata/__init__.py

```diff
@@ -619,23 +788,29 @@ def files(self):
         """
         file_lines = self._read_files_distinfo() or self._read_files_egginfo()
 
-        def make_file(name, hash=None, size_str=None):
+        def make_file(
+            name: str, hash: Optional[str] = None, size_str: Optional[str] = None
+        ) -> PackagePath:
             result = PackagePath(name)
             result.hash = FileHash(hash) if hash else None
             result.size = int(size_str) if size_str else None
             result.dist = self
             return result
 
-        return file_lines and list(starmap(make_file, csv.reader(file_lines)))
+        return (
```

Comment:
Right, so here's a situation where the implementation is perfectly fine - it will always do what we want, but the type system is unable to statically verify that it's so. I was okay with the inelegance of "A and B" because it fit on one line. If this has to break out into multiple lines just to pass through a None value, then I'd probably want a different abstraction to keep this method concise.

Reply:
In 1e19fd2c52b, I've refactored the implementation to retain the concise implementation and to tuck the `None if x is None else` into a decorator from `jaraco.functools`.